client/jetpack-cloud/sections/settings/controller.tsx

```diff
@@ -18,7 +23,25 @@ export const settings: PageJS.Callback = ( context, next ) => {
 
 export const advancedCredentials: PageJS.Callback = ( context, next ) => {
 	const { host, action } = context.query;
-	context.primary = <AdvancedCredentials action={ action } host={ host } role="main" />;
+	const siteId = getSelectedSiteId( context.store.getState() ) as number;
+	const sectionElt = <AdvancedCredentials action={ action } host={ host } role="main" />;
+
+	context.primary = (
+		<HasSiteCredentialsSwitch
```

Comment:
The site credentials and purchases requests should ideally be done concurrently, but I favoured having 2 separate switches here for clarity. It seems that credentials data are often already in the state at this point, so the gain of doing that seems to be minimal.